webgrid/renderers.py

```diff
@@ -703,15 +704,34 @@ class XLSX(object):
     def __init__(self, grid):
         self.grid = grid
         self.styles_cache = LazyDict()
+        self._xlsx_format_cache = {}
         self.default_style = {}
         self.col_widths = {}
 
+    def get_xlsx_format(self, wb, style_dict):
+        """
+        This method is meant to solve a major performance issue with how xlsxwriter manages formats.
+        Xlsxwriter maintains a cache of formats, however generating the cache key is surprisingly
+        expensive since it must join together every property of the format.
+
+        The upshot of this is that if we have several columns with identical style properties but
+        separate xlsxwriter Format objects, the cache key will have to be generated multiple times
+        per cell. It is much faster to use the same Format object for all columns sharing the same
+        style properties.
+
+        See xlsxwriter.format::Format._get_xf_index for how the caching works.
+        """
+        key = tuple(sorted(style_dict.values(), key=itemgetter(0)))
```

Comment:
Something is a little weird about this.

Don't you want the key:value pair to be "key"? Might want to use `items`